### PR TITLE
HPCC-27124 Add Ubuntu and OSX support for vcpkg build 

### PR DESCRIPTION
**Pull request by GordonSmith**
Remove old "Find" modules
Refactor commonSetup.cmake
Fix some minor cmake warnings

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: cmake_modules/vcpkgSetup.cmake [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    #         "name": "libmysql",
+    #         "platform": "!(windows & x86) & !uwp"
+    #     },
+    #     "gtest",
+    #     "curl",
+    #     "curlpp"
+    # ]
+
+#  cmake .. -G "Visual Studio 16 2019" -T host=x64 -A x64 -DCMAKE_TOOLCHAIN_FILE=../vcpkg/scripts/buildsystems/vcpkg.cmake
+# cmake --build . --config Debug -- -m
```
Comment (GordonSmith) [inline cmake_modules/vcpkgSetup.cmake:58]: TODO:  Tidy Up

File: system/security/cryptohelper/digisign.cpp [hunk 2]
```diff
@@ -206,7 +206,9 @@ class CDigitalSignatureManager : public CSimpleInterfaceOf<IDigitalSignatureMana
 
 
 static IDigitalSignatureManager * dsm = nullptr;
+#if defined(_USE_OPENSSL) && !defined(_WIN32)
```
Comment (richardkchapman) [inline system/security/cryptohelper/digisign.cpp:209]: Why is the second condition present? I get that the first is to suppress "unused static variable" errors...

**Review by ghalliday — APPROVED** [review]
Looks sensible to me.  What effect does this have on the existing client builds?

File: .github/workflows/build-vcpkg.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      - name: OSX Dependencies
+        if: ${{ matrix.os == 'macos-10.15' }}
+        shell: "bash"
+        run: |
+          brew install bison flex
+          export CMAKE_INCLUDE_PATH="/usr/local/opt/flex/include"
+          export CMAKE_LIBRARY_PATH="/usr/local/opt/flex/lib;/usr/local/opt/bison/lib"
+          export PATH="/usr/local/opt/flex/bin:/usr/local/opt/bison/bin:$PATH"
+
+          brew install automake libtool cppunit boost mysql-client openldap
```
Comment (richardkchapman) [inline .github/workflows/build-vcpkg.yml:93]: Should add pkg-config to this list I think - at least, I seemed to need it when testing on my mac.

File: system/security/cryptohelper/digisign.cpp (hunk 2 above)
Comment (GordonSmith) [reply system/security/cryptohelper/digisign.cpp:209]: It was matching the existing logic on line 220 in the same file.

Comment (richardkchapman) [reply system/security/cryptohelper/digisign.cpp:209]: It's the logic on line 239 it needs to match, though that also mentions WIN32 for reasons I don't really follow

File: .github/workflows/build-vcpkg.yml (hunk 3 above)
Comment (GordonSmith) [reply .github/workflows/build-vcpkg.yml:93]: Yes (while it is pre installed on the GH Build Machine, having it here will help with the "self documenting" aspect)

File: system/security/cryptohelper/digisign.cpp (hunk 2 above)
Comment (GordonSmith) [reply system/security/cryptohelper/digisign.cpp:209]: There are a few more places its referenced as well.
Also the Windows / OSX version of openssh is currently a different version to the Ubuntu one, which is why I would rather push this off to later.

**Comment by GordonSmith** [comment]
@richardkchapman this can be pulled?

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**